controller/deploy.go

```diff
@@ -60,26 +64,51 @@ func (d Deploy) buildConfigExists(name string) (bool, error) {
 }
 
 func (d Deploy) create(dc DeployCmd) (*DeployResponse, error) {
-	labels := map[string]string{
-		"rmmap/guid":    dc.CloudAppGUID(),
-		"rhmap/project": dc.Project(),
-		"rhmap/domain":  dc.DomainName(),
+	dc.SetAppTag("redis")
+	d.createRedis(dc)
+	dc.AddEnvVar("REDIS_HOST", dc.AppName())
+	dc.SetAppTag("cloud")
+	return d.createCloudApp(dc)
+}
+
+func (d Deploy) createRedis(dc DeployCmd) error {
+	if err := d.paaSService.CreateService(dc, "redis app", 8001); err != nil {
+		return err
+	}
+	if err := d.paaSService.CreateDeploymentConfig(dc, "docker.io/rhmap/redis:2.18.22", "redis deployment config"); err != nil {
+		return err
 	}
-	if _, err := d.paaSService.CreateService(dc.EnvironmentName(), dc.ServiceName(), dc.CloudAppName(), "rhmap cloud app", 8001, labels); err != nil {
+	return nil
+}
+
+func (d Deploy) createCloudApp(dc DeployCmd) (*DeployResponse, error) {
+	if err := d.paaSService.CreateService(dc, "rhmap cloud app", 8001); err != nil {
 		return nil, err
 	}
-	if err := d.paaSService.CreateRoute(dc.EnvironmentName(), dc.ServiceName(), dc.CloudAppName(), "", labels); err != nil {
+	if err := d.paaSService.CreateRoute(dc, ""); err != nil {
 		return nil, err
 	}
-	if err := d.paaSService.CreateImageStream(dc.EnvironmentName(), dc.CloudAppName(), labels); err != nil {
+	if err := d.paaSService.CreateImageStream(dc); err != nil {
 		return nil, err
 	}
 	//create secrets
+	if err := d.paaSService.CreateSecret(dc, map[string][]byte{
+		"username": []byte(dc.UserName()),
+		"password": []byte(dc.Authentication()),
+	}); err != nil {
+		return nil, err
+	}
 
 	//create build config
-	if err := d.paaSService.CreateBuildConfig(dc.EnvironmentName(), dc.BuildConfigName(), dc.CloudAppName(), "rhmap cloud app", dc.SourceLoc(), dc.SourceBranch(), labels); err != nil {
+	if err := d.paaSService.CreateBuildConfig(dc, "Build Config for RHMAP cloud app", "openshift", "nodejs:4"); err != nil {
```

Comment:
I actually thought this should come in from the request ultimately, so supercore can specify other versions of node? I have captured that in a jira.